packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -498,6 +508,23 @@ export class DataStores implements IDisposable {
         return builder.getSummaryTree();
     }
 
+    /**
+     * Before GC runs, called by the garbage collector to update any pending GC state.
+     * The garbage collector needs to know all outbound references that are added. Since root data stores are not
+     * explicitly marked as referenced, notify GC of new root data stores that were added since the last GC run.
+     */
+    public async updateStateBeforeGC(): Promise<void> {
+        for (const id of this.dataStoresSinceLastGC) {
```

Comment:
Any reason this kind of processing is not done directly in processAttachMessage()?
I.e. do we have to delay calling this.rootDataStoreAdded ?

Reply:
We need to find if this is a root data store or not. That information is present in the attributes blob which we have to download asynchronously.

Reply:
I wonder how these things will change with aliasing work. I assume we will need to at least inspect all usages of isRoot and make determination what to do with them, given that "rootiness" will be decoupled from creation process.

Reply:
Let me talk to @andre4i about this.